### PR TITLE
Wait until we receive a response to prompt next override

### PR DESCRIPTION
**Pull request by vinistock**
When users selected decide each override, we were prompting for their decision in a loop inside an async function. This made it so that we triggered all prompts at the same time, but VS Code caps the amount of displayed pop ups at 3. This means that, if we had more than 3 settings to decide overrides for, we would simply lose the prompts for all settings other than the last 3.

This PR changes the loop for a recursive function to check each setting. In this configuration, we only prompt the next override once the user has decided for the current one.

Basically:
- If we're doing `force` or if we know we can override already, we just update the setting and keep going until the end
- If we need to check each setting, only advance the loop and prompt for the next setting when we received a response for the previous one.

The experience change is that users will get prompted for one override at a time, instead of getting spammed with multiple popups.

I tried implementing this in a loop, without recursion, but I couldn't find a decent way of moving to the next iteration only after the promise is resolved.

### CODE REVIEW

File: src/configuration.ts [hunk 1]
```diff
@@ -100,18 +100,36 @@ export class Configuration {
...
-      }
-    });
+    const { scope, section, name, value } = DEFAULT_CONFIGS[settingIndex];
+    const config = this.configurationStore.getConfiguration(section, scope);
+
+    if (canUpdate) {
+      // If we can update, then just update until we exhaust the list
+      config.update(name, value, true, true);
+      this.recursivelyCheckSetting(settingIndex + 1, canUpdate);
+    } else {
```
Comment (KaanOzkan) [inline src/configuration.ts:122]: One way to reduce the recursion is going with 2 methods. For the case when `canUpdate == true` we can do a simple loop. Else, we trigger another method that utilizes recursion.

I don't have strong opinions. It currently reads okay, I'm just not a big fan of `canUpdate` being a flag that is never updated. Having 2 methods would get rid of that. What do you prefer?

**Review by KaanOzkan — APPROVED** [review]
These prompts are crucial so I think it's good to be slightly annoying upon first installation and make sure they're looked at.

**Comment by vinistock** [comment]
Agreed! That's why we can't have VS Code swallowing them up.
